lib/reconstruct.ts

```diff
@@ -0,0 +1,73 @@
+import { types, Root, Token, Tokens, Group } from './types';
+import * as sets from './sets';
+
+const simplifications: [sets.SetFunc, string][] = [
+  [sets.words, '\\w'],
+  [sets.notWords, '\\W'],
+  [sets.ints, '\\d'],
+  [sets.notInts, '\\D'],
+  [sets.whitespace, '\\s'],
+  [sets.notWhitespace, '\\S'],
+  [sets.anyChar, '.']
+];
+
+const reduceStack = (stack: Token[]): string => stack.map(reconstruct).join('');
+
+const createAlternate = (token: Root | Group): string => {
+  if ('options' in token) {
+    return token.options.map(reduceStack).join('|');
+  } else if ('stack' in token) {
+    return reduceStack(token.stack);
```

Comment:
Modifying `Object.prototype` is a bad thing for a user of this library to do. The JavaScript standard is also unlikely to gain `Object.prototype` properties named `options` or `stack`. Still, I think `type` is the safer and better check here.

```suggestion
  if (token.type === types.ROOT) {
    return token.options.map(reduceStack).join('|');
  } else if (token.type === types.GROUP) {
    return reduceStack(token.stack);
```

Reply:
This can become 
```ts
const createAlternate = (token: Root | Group): string => {
  if (token.options) {
    return token.options.map(reduceStack).join('|');
  } else if (token.stack) {
    return reduceStack(token.stack);
  } else {
    throw new Error(`options or stack must be Root or Group token`);
  }
};
```
however, the suggested change doesn't work because the root can have a stack https://github.com/fent/ret.js/blob/47e88c3194641c39b5a82c60a7c6e1fa6a813af1/test/main-test.js#L36-L44

Reply:
Oh, I see, thanks. The issue I was worried about is impossible to fix conveniently and cleanly, then, so it’s probably not worth trying to solve here.